### DISCUSSION TITLE
Repeater field cannot update pivot fields

### DISCUSSION THREAD

**Discussion by yannisc** [filamentphp/filament#10749] [Help] [answered]
I have a repeater field which uses a many-to-many relationship.

It displays data correctly, but when trying to save, I get error Column not found: 1054 Unknown column 'order_commission_rate' in 'field list'. It tries to do this:

update
  `users`
SET
  `order_commission_rate` = 2,
  `users`.`updated_at` = 2024 -01 -10 23: 54: 20
WHERE
  `id` = 4

while it should try to update order_commsion_rate on the order_user pivot table.

I have set the relationship:

    public function sellers()
    {
        return $this->belongsToMany(User::class, 'order_user')
            ->withPivot('order_commission_rate', 'ontotal_amount', 'due_amount')
            ->withTimestamps();
    }

and here is the repeater field:

Repeater::make('sellers')
                        ->relationship('sellers')
                        ->schema([
                            Select::make('user_id')
                            ->options(
                                User::where('enabled', true)->get()->mapWithKeys(function ($user) {
                                    return [$user->id => $user->first_name . ' ' . $user->last_name];
                                    })
                                    ),
                            TextInput::make('order_commission_rate')
                        ])

Please help!

**Comment by yannisc** [comment]
To solve this, I created a new relationship directly to the pivot table and used this:

public function sellers_commissions()
    {
        return $this->hasMany(OrderUser::class, 'order_id');
    }

I think though that the repeater was not meant to be used in that way. Am I getting it right?

**Reply by flashmediasolutions** [reply, marked as answer]
https://filamentphp.com/docs/3.x/forms/fields/repeater#integrating-with-a-belongstomany-eloquent-relationship

**Reply by yannisc** [reply]
thank you!!
